manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -605,13 +638,17 @@ func (r *FybrikApplicationReconciler) updateProvisionedStorageStatus(application
 }
 
 func (r *FybrikApplicationReconciler) buildSolution(applicationContext *api.FybrikApplication, requirements []DataInfo) (map[string]NewAssetInfo, *api.PlotterSpec, error) {
+	uuid := utils.GetFybrikApplicationUUID(applicationContext)
+
 	// get deployed modules
-	moduleMap, err := r.GetAllModules()
+	moduleMap, err := r.GetAllModules(uuid)
```

Comment:
uuid should be a member field of the controller and be updated on each reconcile, rather than be passed to all functions.

Reply:
@shlomitk1 , can the same controller (`FybrikApplicationReconciler`) be concurrently called to reconcile events from different fybrik apps ?

Reply:
It can. Right, uuid can't be a global member then.